app/models/detector/citation.rb

```diff
@@ -135,13 +138,25 @@ def commas(phrase)
       phrase.count(',')
     end
 
+    # This converts the already-built @subpatterns and @summary instance variables into the @features instance variable,
+    # which has a format suitable for sending to our prediction algorithm.
+    def extract_features
+      # Need to create a separate instance variable, so use .deep_dup
+      @features = @subpatterns.deep_dup
+      # Convert the @subpattern structure of {no: = [], pages: ['194-204']} (a hash of matched substrings, with some
+      # empty) into {no: 0, pages: 1} (a hash of integers, some zero)
+      @features = @features.transform_values(&:length)
+      # Now join the re-shaped hash with the @summary variable, so everything is in one place.
+      @features = @features.merge(summary)
+    end
+
     # This builds one of the two main components of the Citation detector - the subpattern report. It uses each of the
     # regular expressions in the CITATION_PATTERNS constant, extracting all matches using the scan method.
     #
     # @return hash
     def pattern_checker(phrase)
       CITATION_PATTERNS.each_pair do |type, pattern|
-        @subpatterns[type.to_sym] = scan(pattern, phrase) if scan(pattern, phrase).present?
+        @subpatterns[type.to_sym] = scan(pattern, phrase)
```

Comment:
Did you find the previous conditional did nothing useful here? I have no concerns specifically with removing the conditional but I'm curious if you realized it was never needed or if something changed to no longer need it.